core/src/main/java/io/questdb/cutlass/http/processors/TextQueryProcessor.java

```diff
@@ -302,9 +302,14 @@ private LogRecord error(TextQueryProcessorState state) {
         return LOG.error().$('[').$(state.getFd()).$("] ");
     }
 
-    protected void header(HttpChunkedResponseSocket socket) throws PeerDisconnectedException, PeerIsSlowToReadException {
+    protected void header(HttpChunkedResponseSocket socket, TextQueryProcessorState state) throws PeerDisconnectedException, PeerIsSlowToReadException {
         socket.status(200, "text/csv; charset=utf-8");
-        socket.headers().put("Content-Disposition: attachment; filename=\"questdb-query-").put(clock.getTicks()).put(".csv\"").put(Misc.EOL);
+        if (state.fineName != null && state.fineName.length() > 0) {
+            socket.headers().put("Content-Disposition: attachment; filename=\"" + state.fineName).put(".csv\"").put(Misc.EOL);
```

Comment:
`header` is a StringSink. Please use `put` method instead of string concatenation